api/task_get.go

```diff
@@ -0,0 +1,29 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/consul-terraform-sync/logging"
+)
+
+// GetTaskByName retrieves a task's information by the task's name
+func (h *TaskLifeCycleHandler) GetTaskByName(w http.ResponseWriter, r *http.Request, name string) {
+	h.mu.Lock()
```

Comment:
Good question, thanks for asking! I lean towards locking since I can imagine a potential race with other API calls. With our current API options, I don't think there will be many writes and so don't imagine a serious contention for the lock. I did update this to be a read lock, which I should have done in the first place. Let me know if you have thoughts.

Reply:
yep, RLock makes way more sense, thanks for making the change Lorna!